lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -95,10 +91,13 @@ defmodule ExUnit.Server do
       when is_integer(loaded) do
     state =
       if uniq? do
+        async_modules = :queue.to_list(state.async_modules) |> Enum.uniq() |> :queue.from_list()
+        sync_modules = :queue.to_list(state.sync_modules) |> Enum.uniq() |> :queue.from_list()
```

Comment:
Not really pretty

Reply:
btw @josevalim speaking about reducing complexity, I've seen that you changed the server to append tests instead of prepending them in https://github.com/elixir-lang/elixir/commit/b665ddd6db823bf24ec039087fdda8367ba27b58, so the changes in ExUnit.Server are not necessary any more. The `:queue` may be a little bit faster, so keep it or not?

Reply:
Did I? I think the code is still prepending, since &1 (the value in state) is on the right side?

Reply:
well never mind, you're right of course